### PR TITLE
Create globally unique Header IDs when marked parses.

### PR DESCRIPTION
**Pull request by dbolack-ab**
Integrates corrections for Document-level unique Header IDs with marked-gfm-header-id updates.

Requires https://github.com/markedjs/marked-gfm-heading-id/issues/542 and an assumed version bump to work.

Validated locally.

Solves #1430 

---

Test code steps from G-Ambatte:

Code:
```
# h1 

\page

# h1
```

On Production:
```
<div class="page" id="p2"><div class="columnWrapper"><h1 id="h1">h1</h1>
<p>&nbsp;</p>
<div class="columnSplit">  </div><p>&nbsp;</p>
</div></div>
```

In Deployment:
```
<div class="page" id="p2"><div class="columnWrapper"><h1 id="h1-1">h1</h1>
<p>&nbsp;</p>
<div class="columnSplit">  </div><p>&nbsp;</p>
</div></div>
```

In Production, the ID for both `<h1>` elements is the same, as the list of existing IDs resets on every page.
In the Deployment, the list of IDs does not reset between pages, so the ID of the second `<h1>` element is set to `h1-1`

### CODE REVIEW

**Comment by dbolack-ab** [comment]
Note: This cannot be tested through automation until the mentioned PR is merged. If you want to test it locally, here is the branch.

https://github.com/dbolack-ab/marked-gfm-heading-id/tree/recurringSlugs

**Comment by dbolack-ab** [comment]
Upstream dependency has been merged. Need to wait for version update...

**Comment by 5e-Cleric** [comment]
bump version PR is here: https://github.com/naturalcrit/homebrewery/pull/3528

**Comment by 5e-Cleric** [comment]
This should work?

**Comment by dbolack-ab** [comment]
> This should work? Unsure on how to test, please update initial comment to explain how to test this PR.

G-Ambette added solid steps.

**Review by G-Ambatte — COMMENTED** [review]
I suspect that by re-evaluating the default value of the `pageNumber` parameter of the `Markdown.render()` function, we can potentially remove ~90% of the current changes in this PR.

File: shared/naturalcrit/markdown.js [hunk 1]
```diff
@@ -818,9 +818,12 @@ let globalPageNumber = 0;
 module.exports = {
 	marked : Marked,
 	render : (rawBrewText, pageNumber=1)=>{
```
[the remark this answers is not in the document]
Comment (dbolack-ab) [reply shared/naturalcrit/markdown.js:820]: I am confident I set it up that way for a reason, but testing and memory tell me either that is wrong or the reason was invalid.

**Comment by calculuschild** [comment]
I think this looks good. Just need to resolve the merge conflicts. I can do this later when I am off work or you can @dbolack-ab .

Easiest way I think would be to just copy `package-lock.json` from master.

**Review by calculuschild — APPROVED** [review]

**Review by 5e-Cleric — APPROVED** [review]

**Comment by 5e-Cleric** [comment]
Nice changes! I fixed the conflict myself, happy merge season!

**Pull request merged**